kolibri/plugins/document_epub_render/assets/src/views/EpubRendererIndex.vue

```diff
@@ -467,6 +520,57 @@
       delete global.ePub;
     },
     methods: {
+      // TODO: Get rid of this when Firefox breaks tables properly in CSS column layout
+      splitTable(maxHeight, table) {
+        // Chunk all rows into groups that will fit into a table within `maxHeight`
+        const tableSets = Array.from(table.getElementsByTagName('tr')).reduce(
+          (tableSets, row) => {
+            const current = tableSets.pop();
+            const currentHeight = current.reduce((h, row) => h + row.clientHeight, 0);
+
+            if (current.length && currentHeight + row.clientHeight >= maxHeight) {
+              tableSets.push(current);
+              tableSets.push([row]);
+              return tableSets;
+            }
+
+            current.push(row);
+            tableSets.push(current);
+            return tableSets;
+          },
+          [[]]
+        );
+
+        // Loop through row chunks and create the replacement tables
+        return tableSets.map(tableSet => {
+          // Deep clone to maintain as much as possible
+          const newTable = table.cloneNode(true);
+
+          // Then remove all the unnecessary leftovers
+          const toRemove = [
+            ...Array.from(newTable.getElementsByTagName('thead')),
+            ...Array.from(newTable.getElementsByTagName('tfoot')),
+            ...Array.from(newTable.getElementsByTagName('tr')),
+          ];
+
+          toRemove.forEach(el => {
+            el.parentElement.removeChild(el);
+          });
+
+          const tbody = newTable.getElementsByTagName('tbody')[0];
+
+          // Shouldn't happen :/
+          if (!tbody) {
+            return newTable;
+          }
+
+          tableSet.forEach(row => {
+            tbody.appendChild(row);
+          });
+
+          return newTable;
+        });
+      },
```

Comment:
not critical, but might be nice to quarantine these two big code blocks in an external `firefoxTableHack.js` file or similar?

Reply:
That's reasonable. They are large chunks of meaty code. I'll do that.